checkov/common/graph/checks_infra/enums.py

```diff
@@ -0,0 +1,9 @@
+from enum import Enum
+
+
+class SolverType(str, Enum):
+    ATTRIBUTE = "ATTRIBUTE"
+    COMPLEX = "COMPLEX"
+    CONNECTION = "CONNECTION"
+    COMPLEX_CONNECTION = "COMPLEX_CONNECTION"
```

Comment:
Please add a comment expalining what a "solver". And a comment explaining what each solver type is

Reply:
Modified in https://github.com/bridgecrewio/checkov/pull/1023/commits/aaa1d1574bda315745fb42e05c7ef24493d10454